### PR TITLE
feat: add Card component

### PR DESCRIPTION
**Pull request by connorlindsey**
Adds the Card component. Will tackle cleaning up shared utils / hooks / etc. after adding the modal. 

CleanShot 2024-10-03 at 15 43 12

### CODE REVIEW

**Comment by connorlindsey** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/knocklabs/javascript/233?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#233** <a href="https://app.graphite.dev/github/pr/knocklabs/javascript/233?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#231** <a href="https://app.graphite.dev/github/pr/knocklabs/javascript/231?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#229** <a href="https://app.graphite.dev/github/pr/knocklabs/javascript/229?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#228** <a href="https://app.graphite.dev/github/pr/knocklabs/javascript/228?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#227** <a href="https://app.graphite.dev/github/pr/knocklabs/javascript/227?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @connorlindsey and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by thomaswhyyou — APPROVED** [review]

File: packages/react/src/modules/in-app-message/components/Card/styles.css [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+.iam-card__action {
+  text-decoration: none;
+  font-size: var(--knock-text-3);
+  font-weight: var(--knock-weight-regular);
+  line-height: var(--knock-leading-3);
+  border-radius: var(--knock-rounded-3);
+  padding: var(--knock-spacing-1) var(--knock-spacing-4);
+  border: 0.5px solid var(--iam-accent);
+  background: var(--iam-accent);
```
Comment (thomaswhyyou) [inline packages/react/src/modules/in-app-message/components/Card/styles.css:56]: Actually one question on this - i thought the buttons are black and white based on krisna's design: https://www.figma.com/design/2SYT1WfJIBUB370HjmLL0N/Message-types?node-id=1570-38052&node-type=section&m=dev

Mind clarifying with him?

File: examples/in-app-message-example/src/App.tsx [hunk 2]
```diff
@@ -80,14 +81,28 @@ function App() {
           </button>
           <hr />
           <h2>Banner</h2>
-          <div>
-            <Banner.Default />
-          </div>
+          <Banner.Default />
```
[the remark this answers is not in the document]
Comment (connorlindsey) [reply examples/in-app-message-example/src/App.tsx:84]: Yep! Got a todo item to reconsider how we export the batteries included vs. presentational components

File: packages/react/src/modules/in-app-message/components/Banner/Banner.tsx [hunk 3]
```diff
@@ -28,7 +28,7 @@ export interface BannerContent {
     text: string;
     action: string;
   };
-  dismissible: boolean;
+  dismissible?: boolean;
```
[the remark this answers is not in the document]
Comment (connorlindsey) [reply packages/react/src/modules/in-app-message/components/Banner/Banner.tsx:31]: I was basing it off of `banner.ex` where required is false (although there is a default value), so I guess we can make this required.

File: packages/react/src/modules/in-app-message/components/Card/styles.css (hunk 1 above)
Comment (connorlindsey) [reply packages/react/src/modules/in-app-message/components/Card/styles.css:56]: Yeah, I'll double check. I was going off of the components in the public figma file: https://www.figma.com/design/eFImkjiK6Pai3mQ3HN2tG7/In-app-Notifications-UI-Kit?node-id=322-2020&node-type=canvas&t=9Bd2q3F3h43N4Xr1-0

Comment (connorlindsey) [reply packages/react/src/modules/in-app-message/components/Card/styles.css:56]: Its meant to be black and white by default! Will update the base theme to match that

Comment (connorlindsey) [reply packages/react/src/modules/in-app-message/components/Card/styles.css:56]: Updated in the modal PR just to make sure it looked alright there as well

**Pull request merged**